### PR TITLE
Fix race condition during disk creation

### PR DESCRIPTION
**Pull request by khenderick**
Removes possiblity of ending up with two disks in the model

### CODE REVIEW

File: ovs/lib/vdisk.py [hunk 1]
```diff
@@ -291,15 +291,20 @@ def clone(vdisk_guid, name, snapshot_id=None, storagerouter_guid=None):
...
-            new_vdisk.size = vdisk.size
+            new_vdisk = VDiskList.get_vdisk_by_volume_id(volume_id)
+            if new_vdisk is None:
+                new_vdisk = VDiskList.get_by_devicename_and_vpool(devicename, vdisk.vpool)
+                if new_vdisk is None:
+                    new_vdisk = VDisk()
+                    new_vdisk.name = VDiskController.extract_volumename(devicename)
+                    new_vdisk.volume_id = volume_id
+                    new_vdisk.size = vdisk.size
+                    new_vdisk.name = name
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py]: Saving the name twice might give unexpected results

File: ovs/lib/vdisk.py [hunk 2]
```diff
@@ -552,14 +557,19 @@ def create_from_template(vdisk_guid, name, storagerouter_guid=None):
...
-            new_vdisk.devicename = devicename
+            new_vdisk = VDiskList.get_vdisk_by_volume_id(volume_id)
+            if new_vdisk is None:
+                new_vdisk = VDiskList.get_by_devicename_and_vpool(devicename, vdisk.vpool)
+                if new_vdisk is None:
+                    new_vdisk = VDisk()
+                    new_vdisk.name = VDiskController.extract_volumename(devicename)
+                    new_vdisk.volume_id = volume_id
+                    new_vdisk.size = vdisk.size
+                    new_vdisk.name = name
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py]: Same here

File: ovs/lib/vdisk.py [hunk 3]
```diff
@@ -617,14 +627,18 @@ def create_new(volume_name, volume_size, storagedriver_guid):
...
-            new_vdisk.save()
+            new_vdisk = VDiskList.get_vdisk_by_volume_id(volume_id)
+            if new_vdisk is None:
+                new_vdisk = VDiskList.get_by_devicename_and_vpool(devicename, vpool)
+                if new_vdisk is None:
+                    new_vdisk = VDisk()
+                    new_vdisk.size = volume_size
+                    new_vdisk.vpool = vpool
+                    new_vdisk.devicename = devicename
+                    new_vdisk.name = volume_name
```
Comment (kvanhijf) [inline ovs/lib/vdisk.py]: No ```extract_volumename``` required here?

**Review by kvanhijf — CHANGES_REQUESTED** [review]

File: ovs/lib/vdisk.py (hunk 3 above)
Comment (khenderick) [reply ovs/lib/vdisk.py]: No, these ```extract_volumename``` code was actually a copy-paste leftover from the ```resize_from_voldrv``` call where - if the disk does not exist yet - it sets a best-guess name.

**Review by kvanhijf — APPROVED** [review]

**Pull request merged**